test/test_logging_macros.cpp

```diff
@@ -108,18 +108,28 @@ TEST_F(TestLoggingMacros, test_logging_expression) {
 }
 
 int g_counter = 0;
+bool g_function_called = false;
 
-bool mod3()
+bool not_divisible_by_three()
 {
+  g_function_called = true;
   return (g_counter % 3) != 0;
 }
 
 TEST_F(TestLoggingMacros, test_logging_function) {
+  // check that evaluation of a specified function does not occur if the severity is not enabled
+  g_rcutils_logging_default_severity_threshold = RCUTILS_LOG_SEVERITY_INFO;
+  RCUTILS_LOG_DEBUG_FUNCTION(&not_divisible_by_three, "should not be logged");
```

Comment:
We still need to set `g_counter` to 1 otherwise `not_divisible_by_three` returns `false`. The check of `g_function_called` is still valid but it makes the check for `g_log_calls` not relevant

Reply:
Really just checking `g_function_called` is `false` _should_ be sufficient here, but more coverage can't hurt: 8c369f940f1ecd72212f8c943092033c13394f2c